Add sslib master tox build + flesh out test docs

lukpueh opened the pull request:
**Fixes issue #**:
None. Suggested by @joshuagl in #913 

**Description of the changes being introduced by the pull request**:
Add a tox build that runs tests against `securesystemslib`'s tip of development, i.e. its `master` branch, to ease preparation of `tuf` for a new `securesystmeslib` release.

The tox build is run on travis but is allowed to fail.

This PR also fleshes out the testing section of the contribution documentation.


**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

Commit:
Add sslib master tox build + flesh out test docs

Add a tox build that runs tests against securesystemslib's tip of
development, i.e. master branch, to ease preparation of tuf for a
new securesystmeslib release.

The tox build is run on travis but is allowed to fail.

This commit also fleshes out the testing section of the
contribution documentation.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

joshuagl commented:
Oh, this looks great. Thanks @lukpueh

lukpueh commented:
@aditya, would you mind taking a quick look at this PR, especially the updated "Testing" section of the contribution doc?

adityasaky commented on docs/CONTRIBUTORS.rst:
```diff
@@ -107,9 +108,46 @@ To work on the TUF project, it's best to perform a development install.
+    $ python aggregate_tests.py
+
+
+To run the tests, measuring code coverage, the script can be run with the
```
Should `coverage` be added to `dev-requirements.txt`?

lukpueh replied:
Good thinking! It probably should be, although `dev-requirements.txt` pins all dependencies, and that IMHO doesn't make sense for a development tool.
 Maybe we can adopt a requirements hierarchy similar to the one proposed in https://github.com/in-toto/in-toto/commit/06a28987dc8092afa1060ee51cd3e19a6f324855. I'd leave that for another PR though.

As a temporary fix we could either add it to `dev-requirements.txt` pinned or unpinned (maybe with a TODO comment), or add a note to `CONTRIBUTORS.rst`, telling contributors to install it separately when running tests with `coverage`. I tend towards the second for above reasons. What do you think?

adityasaky replied:
Agreed, the second option is better as sometimes temporary fixes have a bad habit of becoming permanent!

Commit:
Add extra installation instruction to contrib doc

Add hint to install `coverage` before using it.

This should be installed via dev-requirements.txt, however it
does not seem to fit in there, because dev-requirements.txt pins
all its dependencies which does not seem to make sense for a
development tool.

Maybe a hierarchy of requirements.txt similar to
https://github.com/in-toto/in-toto/commit/06a28987dc8092afa1060ee51cd3e19a6f324855
could be established.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

lukpueh commented:
Thanks for the review, @aditya!
